### DISCUSSION TITLE
Link an external button to an event button (e.g. Charge)

### DISCUSSION THREAD

**Discussion by ArtG07** [artisan-roaster-scope/artisan#1210] [Q&A]
Hello,

I'd like to link an external button, connected either to my raspberry pi 4 or to a HUB0000_0 phidget, to an Artisan event button such as charge, so that when I press it, the associated command is launched.

My question was whether this was possible and, if so, how could I go about it?

Any help will be appreciated, thanks.

**Comment by roasterdave** [comment]
You can use a VINT port.  Refer to Section 3 on this page for port information. https://artisan-scope.org/devices/phidgets/  

Create a **Phidget HUB IO Digital** device for the port where the button is connected (Config>> Device>> Extra Devices tab).

Then create a new alarm (Config>> Alarms).  Set the **Source** to be the extra device just created.  The **Phidget HUB IO Digital** device's value will be 0 or 1, for open or closed, so set the alarm **Value** to 0.5 and the **Condition** to "above" or "below" based on the type of switch (NO or NC).  Set the alarm **Action** to the desired event, such as "CHARGE".

Remember that only one software at a time can access the Phidgets so the Phidget Control Panel must be closed when using Artisan.

![1210-1](https://github.com/artisan-roaster-scope/artisan/assets/30009312/36fe16d2-448d-4e27-95f3-f4fa68178d0a)
![1210-2](https://github.com/artisan-roaster-scope/artisan/assets/30009312/614015be-e225-410e-bef8-9c5c4daa3fe5)

**Comment by MAKOMO** [comment]
Alternatives are to use a [Steam Deck](https://www.home-barista.com/roasting/keypad-control-artisan-t82836.html) or to use the SPACE key to enter events using the [Artisan keyboard mode introduced in Artisan v0.99](https://artisan-roasterscope.blogspot.com/2016/03/artisan-v099.html)

**Comment by ArtG07** [comment]
Thank you both for your prompt replies, I'm very grateful.

I must admit that the Stream Deck solution was a good one, but unfortunately a bit expensive for what I wanted to do with it.

But the alarm solution works very well, and I hadn't given enough thought to this alarm system, which is super interesting.

Thanks again for your answers, and for this very good solution.
